src/event-manager/event-callback-manager.ts

```diff
@@ -0,0 +1,82 @@
+import EventFilter from './event-filter';
+import Subscription from './subscription';
+import { BlockchainEventTypes, EventConfigType } from '../types';
+import { PushId } from '../ain-db/push-id';
+
+export default class EventCallbackManager {
+  private readonly _filters: { [filterId: string]: EventFilter };
+  private readonly _filterIdToSubscriptionId: { [filterId: string]: string };
+  private readonly _subscriptions: { [subscriptionId: string]: Subscription };
+
+  constructor() {
+    this._filters = {};
+    this._filterIdToSubscriptionId = {};
+    this._subscriptions = {};
+  }
+
+  buildFilterId() {
+    return PushId.generate();
+  }
+
+  buildSubscriptionId() {
```

Comment:
I agree. However, filter and subscription seemed to have different levels, so they were separated. 😭  It didn't look good to have different types of objects with the same Id.

Reply:
Well.., do we need subscription id from the beginning? (you mentioned that we're trying to hide information from the end users as much as possible, which means the end users won't be using subscription ids?)
it looks like we're looking up subscriptions in two steps: 1) filter id -> subscription id, 2) subscription id -> subscription.
I think this can be simplified by replacing _filterIdToSubscriptionId with _filterIdToSubscription and not using subscription ids?

Reply:
hmm. Looks good!

Reply:
Done